src/components/RepetitionsExercises.jsx

```diff
@@ -0,0 +1,39 @@
+import {useMemo} from 'react';
+
+import useStore from '../hooks/useStore';
+
+import ExerciseCard from './ExerciseCard';
+import PauseCard from './PauseCard';
+
+export default function RepetitionsExercises() {
+	const currentExercise = useStore(state => state.currentExercise);
+
+	const amountRepetitions = currentExercise.repetition;
+
+	const exercises = useMemo(() => {
+		const _exercises = [];
+		for (let i = 0; i < amountRepetitions; i++) {
+			_exercises.push(() => (
+				<div>
+					<ExerciseCard />
+					{i < amountRepetitions - 1 &&
+					(currentExercise.pause.minutes > 0 || currentExercise.pause.seconds > 0) ? (
+						<PauseCard />
+					) : (
+						''
+					)}
+				</div>
+			));
+		}
+
+		return _exercises;
+	}, [amountRepetitions, currentExercise]);
```

Comment:
Preparing the whole output in a variable actually looks a bit strange to me. Isn't there a way to put all of this in the return of `RepetitionsExercises()`? Like put the whole `for` loop directly in the return, than you don't need the variables `exercises` and `_exercises`, and instead of `_exercise.push` you  return JSX. I'm not sure if this will work though, maybe just try it?

Reply:
It wont work i think because i cant use a useMemo function an a useMemo function thats why i imported it as a component, i tried what you said befor. I wish it would work.

Reply:
I see! That's exactly the reason I thought about.

Reply:
ich glaube, da ich einen array erzeugen muss wo er später wieder durch maped geht das leider nicht, daher brauche ich die variable durch die, die vom for loop generierten daten nochmal gemapped werden. Oder glaubst du das kann ich anders lösen?

Reply:
Vielleicht habe ich einen Denkfehler, aber meine Idee war, statt einen Array zu erzeugen und später durchzumappen könntest du das was in den Array kommt einfach direkt returnen. Ich visualisiere dir das mal als Pseudocode:

return (
 for (number of reps) {
    return (
     <jsx />) 
  }
)

Also dann noch alles korrekt mit {} escapen und so... aber eigentlich denke ich, das müsste gehen.

Reply:
Also, wie gesagt, wenn du weißt, dass das nicht geht, dann ist das so. Kann gut sein, dass ich beim Draufblicken von außen etwas ganz furchtbar übersehe. Wenn du einfach weitermachen willst, geb ich dir auch gern einen grünen Hacken. :-)

Reply:
ich werde mir das heute Abend oder morgen früh nochmal genauer ansehen und probieren, geb dir dann nochmal ein feedback ;) brauche bis montag keinen Grünen hacken alles gut. Schönes Wochenende dir und wir hören uns :)

Reply:
Bis Montag schafefn wir. Hab du auch ein schönes Wochenende!